test/jdk/sun/java2d/cmm/ColorConvertOp/FilterSemiCustomImages.java

```diff
@@ -0,0 +1,161 @@
+            ColorSpace.CS_PYCC, ColorSpace.CS_sRGB
+    };
+
+    private static final class CustomRaster extends WritableRaster {
+        CustomRaster(SampleModel sampleModel, Point origin) {
+            super(sampleModel, origin);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        for (int fromIndex : CSS) {
+            for (int toIndex : CSS) {
+                if (fromIndex != toIndex) {
+                    for (int type : TYPES) {
+                        test(fromIndex, toIndex, type);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void test(int fromIndex, int toIndex, int type)
+            throws Exception
+    {
+        ColorSpace fromCS = ColorSpace.getInstance(fromIndex);
+        ColorSpace toCS = ColorSpace.getInstance(toIndex);
+        ColorConvertOp op = new ColorConvertOp(fromCS, toCS, null);
+
+        // standard source -> standard dst
+        BufferedImage srcGold = new BufferedImage(W, H, type);
+        fill(srcGold);
+        BufferedImage dstGold = new BufferedImage(W, H, type);
+        op.filter(srcGold, dstGold);
+
+        // custom source -> standard dst
+        BufferedImage srcCustom = makeCustomBI(srcGold);
+        fill(srcCustom);
+        BufferedImage dst = new BufferedImage(W, H, type);
+        op.filter(srcCustom, dst);
+        verify(dstGold, dst);
+
+        // standard source -> custom dst
+        BufferedImage src = new BufferedImage(W, H, type);
+        fill(src);
+        BufferedImage dstCustom = makeCustomBI(dstGold);
+        op.filter(src, dstCustom);
+        verify(dstGold, dstCustom);
+
+        // custom source -> custom dst
+        srcCustom = makeCustomBI(srcGold);
+        fill(srcCustom);
+        dstCustom = makeCustomBI(dstGold);
+        op.filter(srcCustom, dstCustom);
+        verify(dstGold, dstCustom);
+    }
+
+    private static BufferedImage makeCustomBI(BufferedImage bi) {
+        ColorModel cm = bi.getColorModel();
+        SampleModel sm = bi.getSampleModel();
+        CustomRaster cr = new CustomRaster(sm, new Point());
+        return new BufferedImage(cm, cr, bi.isAlphaPremultiplied(), null) {
+            @Override
+            public int getType() {
+                return bi.getType();
+            }
+        };
+    }
+
+    private static void fill(BufferedImage image) {
+        int width = image.getWidth();
+        int height = image.getHeight();
+        for (int x = 0; x < width; ++x) {
+            for (int y = 0; y < height; ++y) {
+                // alpha channel may be calculated slightly differently on
+                // different code paths, so only check fully transparent and
+                // fully opaque pixels
+                Color c = new Color(y * 255 / (height - 1),
+                                    x * 255 / (width - 1),
+                                    x % 255, (x % 2 == 0) ? 0 : 255);
```

Comment:
```suggestion
                Color c = new Color(y * 255 / (height - 1),
                                    x * 255 / (width - 1),
                                    x % 255,
                                    (x % 2 == 0) ? 0 : 255);
```
This way the fourth parameter stands out.